RomanNumeralsCI/src/main/java/com/ericsson/peterbuki/RomanConverter.java

```diff
@@ -4,43 +4,83 @@
  * Created by epetbuk on 2016.05.18..
  */
 public class RomanConverter {
-
+    public static final int MAX_VALUE = 3999 ;
 
     protected RomanConverter()
     {
         System.out.println("RomanConverter constructor.") ;
-        for (RomanNumber n : RomanNumber.values())
+        for (RomanNumberEnum n : RomanNumberEnum.values())
         {
             System.out.println(n.name + " = " + n.value) ;
         }
     }
 
     public static int convertToArabic(String in)
     {
+        boolean wasLastDouble ;
         int result = 0 ;
-        String toDecode ;
-        for (int position = 0; position < in.length(); position++)
+        for (int position = 0; position < in.length(); position+=wasLastDouble?2:1)
         {
-            RomanNumber rn = RomanNumber.Z ;
+            wasLastDouble=false ;
+            RomanNumberEnum rn = RomanNumberEnum.Z ;
 
             // try to convertToRoman 2 characters
             if (in.length()-position > 1)
-                rn = RomanNumber.findByName(in.substring(position, position+2)) ;
+                rn = RomanNumberEnum.findByName(in.substring(position, position+2)) ;
 
             // converted 2 characters, pos increased by additional 1
-            if (rn != RomanNumber.Z)
-                position++ ;
+            if (rn != RomanNumberEnum.Z)
+                wasLastDouble=true ;
             else
-                rn = RomanNumber.findByName(in.substring(position, position+1)) ;
+                rn = RomanNumberEnum.findByName(in.substring(position, position+1)) ;
 
             result += rn.value ;
         }
         return result ;
     }
+    public static boolean isArabic(String in)
+    {
+        boolean result ;
+        try
+        {
+            int number = Integer.parseInt(in) ;
+            result = (number <= MAX_VALUE) ;
+        }
+        catch (NumberFormatException e)
+        {
+            result = false ;
+        }
+        return result;
+    }
+
+    public static boolean isRoman(String in)
+    {
+        for (int pos=0 ; pos < in.length() ; pos++)
+        {
+            System.out.println("Checking: " + in.substring(pos,pos)) ;
+            if (RomanNumberEnum.findByName(in.substring(pos,pos)) == RomanNumberEnum.Z)
```

Comment:
I would use `charAt` instead of `substring(pos,pos)` which means that you may have to change the findByName method as well.